### DISCUSSION TITLE
rux-checkbox should support `required` prop

### DISCUSSION THREAD

**Discussion by nortonprojects** [RocketCommunicationsInc/astro#1056] [Feature Requests]
All of the other form elements support the required prop, as well as the checkboxGroup, but a single checkbox can't be set as required.

**Comment by micahjones13** [comment]
Hey Jacob, 

Right now, the `required` prop on form elements is purely a style change (adds the asterisk). Element internals aren't a part of Stencil yet, so the ValidityState can't be accessed. Once element internals are added, this is definitely something we'll look at. 

I'm going to throw the Stencil element internals ask here just so we can keep track of it:  https://github.com/ionic-team/stencil/issues/2284
